src/compiler/checker.ts

```diff
@@ -13197,10 +13197,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         else if (!member.symbol.isReplaceableByMethod) {
             symbol.declarations.push(member);
         }
-        if (symbolFlags & SymbolFlags.Value) {
-            if (!symbol.valueDeclaration || symbol.valueDeclaration.kind !== member.kind) {

```

Comment:
Was the original logic trying to preserve the somewhat-sketchy logic in the binder around which type of declaration "wins" out?

Reply:
It sure looks like it, but there's a separate utility function for that now: `setValueDeclaration`, which is more complicated*. Specifically, the `kind` check is *only* for namespaces there, which wouldn't apply to `this` property assignments.

I think the better fix to try is

```ts
if (symbolFlags & SymbolFlags.Value) {
  setValueDeclaration(symbol, member)
}
```

In other words, what the binder does in `addDeclarationToSymbol`.

*It also has a 3rd case for function property assignments in .d.ts files.

Reply:
I pushed out this change. Could you take another look?